### PR TITLE
Modify spring-cloud-azure-core, spring-cloud-azure-service, and spring-cloud-azure-resourcemanager according to API review comments

### PR DESCRIPTION
**Pull request by saragluna**
The spring-cloud-azure-core is being reviewed by the architects, this PR is to apply the changes according to comments. 

spring-cloud-azure-core: [spring-cloud-azure-core (version 4.0.0-beta.3) - apiview.dev](https://apiview.dev/Assemblies/Review/eebb74d23c7c462fbdab909e43fb465b)
spring-cloud-azure-service: [spring-cloud-azure-service (version 4.0.0-beta.3) - apiview.dev](https://apiview.dev/Assemblies/Review/98072d31800c4426a5dca9091d266950)
spring-cloud-azure-resourcemanager: [spring-cloud-azure-resourcemanager (version 4.0.0-beta.3) - apiview.dev ](https://apiview.dev/Assemblies/Review/e1da11207fc44c109d78d5d10f697840)

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/properties/AzureEventHubsProperties.java [hunk 1]
```diff
@@ -129,14 +129,14 @@ public Processor getProcessor() {
     /**
      * Properties of an Event Hub producer.
      */
-    public static class Producer extends AzureEventHubsCommonProperties implements EventHubsProducerDescriptor {
+    public static class Producer extends AzureEventHubsCommonProperties implements EventHubProducerProperties {
```
[the remark this answers is not in the document]
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/properties/AzureEventHubsProperties.java:132]: Given Event Hubs only have one client builder, EventHubClientBuilder, so we won't call it EventHubProducerClientProperties.

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**